onnxruntime/contrib_ops/webgpu/quantization/matmul_nbits.cc

```diff
@@ -867,6 +978,40 @@ Status MatMulNBits::ComputeInternal(onnxruntime::webgpu::ComputeContext& context
     return context.RunProgram(mul_program);
   }
 
+  // Block32 prefill program
+  // This program is optimized for Block32 prefill using Tile16x128.
+  const bool use_block32_program = block_size == 32 && batch_count == 1 && !has_zero_points &&
```

Comment:
what was the impact to generation speed ? Should you restrict this shader with a M > kMinMForTileOptimization check.

Reply:
you also support batch size other than 1 and zero points, in your shader perhaps relax that check. Okay to do in a follow up change.